src/components/views/elements/ImageView.tsx

```diff
@@ -302,9 +309,28 @@ export default class ImageView extends React.Component<IProps, IState> {
                 ev.preventDefault();
                 this.props.onFinished();
                 break;
+            case KeyBindingAction.MediaSave:
+                ev.preventDefault();
+                ev.stopPropagation();
+                this.downloadImage();
+                break;
         }
     };
 
+    private async downloadImage(): Promise<void> {
+        const res = await fetch(this.props.src);
+        if (!res.ok) {
+            throw parseErrorResponse(res, await res.text());
```

Comment:
What happens if this throws? I'm not sure the onKeyDown handler will be expecting to get an exception. What will inform the user that it failed?

Reply:
Also, this presumably will be cached / a blob normally, but might it take a while? In which case, should we be showing a spinner?